### PR TITLE
Kajam 11.17.0 plat 5604 html purifier

### PR DESCRIPTION
**Pull request by noam-arad**

### CODE REVIEW

File: admin_console/forms/PartnerConfiguration.php [hunk 1]
```diff
@@ -674,6 +685,8 @@ public function populateFromObject($object, $add_underscore = true)
 				$this->setPermissionGroupElementsToDisabled(array('accountPackagesService'));
 			}
 		}
+
+//		$this->setDefaults(array('html_purifier_behaviour' => $object->htmlPurifierBehaviour));
```
Comment (MosheMaorKaltura) [inline admin_console/forms/PartnerConfiguration.php]: remove obsolete code

File: api_v3/lib/KalturaBaseService.php [hunk 2]
```diff
@@ -148,6 +148,9 @@ public function initService($serviceId, $serviceName, $actionName)
 		}
 		
 		$this->setPartnerFilters($partnerId);
+		
+		$partner = $this->getPartner();
```
Comment (MosheMaorKaltura) [inline api_v3/lib/KalturaBaseService.php]: Use partner object on this object instead

**Comment by MosheMaorKaltura** [comment]
1. Add to partner->create a purifier value set to reject.
2. In case of notify - need to add email notification for case of notify.

File: api_v3/lib/KalturaBaseService.php (hunk 2 above)
Comment (noam-arad) [inline api_v3/lib/KalturaBaseService.php]: I checked, "getPartner" uses the partner field

File: alpha/lib/model/Partner.php [hunk 3]
```diff
@@ -1757,4 +1759,14 @@ public function setUsageWarning($type, $percent, $value){
 		$usageWarnings[$type.'_'.$percent] = $value;
 		$this->setUsageWarnings($usageWarnings);		
 	}
+	
+	public function getHtmlPurifierBehaviour()
+	{
+		return $this->getFromCustomData( self::HTML_PURIFIER_BEHAVIOUR, null , HTMLPurifierBehaviourType::IGNORE );
```
Comment (yossipapi) [inline alpha/lib/model/Partner.php]: I don't see anywhere in the code a definition of HTMLPurifierBehaviourType

File: alpha/apps/kaltura/lib/myPartnerRegistration.class.php [hunk 4]
```diff
@@ -252,6 +252,8 @@ private function createNewPartner( $partner_name , $contact, $email, $ID_is_for,
 	 	if ($newPartner->getType() == Partner::PARTNER_TYPE_WORDPRESS)
 	 		kPermissionManager::setPs2Permission($newPartner);
 	 	
+		$newPartner->setHtmlPurifierBehaviour(HTMLPurifierBehaviourType::BLOCK);
```
Comment (yossipapi) [inline alpha/apps/kaltura/lib/myPartnerRegistration.class.php]: I don't see anywhere in the code the class HTMLPurifierBehaviourType

File: alpha/lib/model/Partner.php (hunk 3 above)
Comment (noam-arad) [inline alpha/lib/model/Partner.php]: please have a look at KalturaSystemPartnerConfiguration.php:341

File: alpha/apps/kaltura/lib/myPartnerRegistration.class.php (hunk 4 above)
Comment (noam-arad) [inline alpha/apps/kaltura/lib/myPartnerRegistration.class.php]: please have a look at KalturaSystemPartnerConfiguration.php:341

**Pull request merged**